IdentityCore/src/broker_operation/request/token_request/MSIDBrokerOperationBrowserTokenRequest.m

```diff
@@ -0,0 +1,99 @@
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDBrokerOperationBrowserTokenRequest.h"
+#import "MSIDJsonSerializableTypes.h"
+#import "MSIDAADAuthority.h"
+
+@implementation MSIDBrokerOperationBrowserTokenRequest
+
+- (instancetype)initWithRequest:(NSURL *)requestURL
+                        headers:(NSDictionary *)headers
+               bundleIdentifier:(NSString *)bundleIdentifier
+                          error:(NSError **)error
+{
+    self = [super init];
+    if (self)
+    {
+        if (![requestURL isKindOfClass:[NSURL class]])
+        {
+            if (error)
+            {
+               NSString *errorMessage = [NSString stringWithFormat:@"Failed to create browser operation request due to invalid request url %@",requestURL];
+               *error = MSIDCreateError(MSIDErrorDomain,MSIDErrorInvalidInternalParameter,errorMessage,nil, nil, nil, nil, nil, YES);
+            }
+                  
+           return nil;
+        }
+        
+        _requestURL = requestURL;
+        
+        if (![self isAuthorizeRequest:_requestURL])
+        {
+            if (error)
+            {
+                NSString *errorMessage = [NSString stringWithFormat:@"Failed to create browser operation request, %@ is not authorize request", [requestURL absoluteString]];
+                *error = MSIDCreateError(MSIDErrorDomain,MSIDErrorInvalidInternalParameter,errorMessage,nil, nil, nil, nil, nil, YES);
+            }
+                   
+            return nil;
+        }
+        
+        _headers = headers;
+        _bundleIdentifier = bundleIdentifier;
+        
+        MSIDAADAuthority *authority = [[MSIDAADAuthority alloc] initWithURL:_requestURL rawTenant:nil context:nil error:error];
+        
+        if (!authority)
+        {
+            return nil;
+        }
+        
+        _authority = authority;
+        
+        _correlationId = [NSUUID UUID];
+    }
+    
+    return self;
+}
+
+- (BOOL)isAuthorizeRequest:(NSURL *)url
+{
+    NSString *request = [url absoluteString];
+    BOOL isAuthorizeRequest = ([request rangeOfString:@"/oauth2/authorize?" options:NSCaseInsensitiveSearch].location != NSNotFound);
+    BOOL isV2AuthorizeRequest = ([request rangeOfString:@"/oauth2/v2.0/authorize?" options:NSCaseInsensitiveSearch].location != NSNotFound);
```

Comment:
I'd recommend removing "?" from matching to avoid mismatches if someone is sending not perfectly correct, but otherwise working URL.

Reply:
Addressed.